src/overlays/actors/ovl_En_Stream/z_en_stream.c

```diff
@@ -30,29 +29,121 @@ const ActorInit En_Stream_InitVars = {
     (ActorFunc)EnStream_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80965B20[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 20, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_80965B20[];
-
-extern UNK_TYPE D_06000950;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/EnStream_SetupAction.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/EnStream_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/EnStream_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/func_809656D4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/func_809657F4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/func_8096597C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/EnStream_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/EnStream_Draw.s")
+extern Gfx D_06000950[];
+
+void EnStream_SetupAction(EnStream* this, EnStreamActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnStream_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnStream* this = THIS;
+
+    this->size = EN_STREAM_SIZE(&this->actor);
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    if (this->size != EN_STREAM_SIZE_NORMAL && this->size == EN_STREAM_SIZE_SMALL) {
+        this->actor.scale.y = 0.01f;
+    }
+    EnStream_SetupAction(this, EnStream_WaitForPlayer);
+}
+
+void EnStream_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+s32 EnStream_PlayerIsInRange(Vec3f* vortexPosRot, Vec3f* playerPosRot, Vec3f* posDifference, f32 vortexYScale) {
```

Comment:
This looks like a bool, but has an extra case, so you should `!= 0` it in the ifs where it's used. It should maybe get an enum too, not clear precisely how different the cases are.

Reply:
Good point. Also, the names for the arguments to this function are terrible, so let me fix those too.